### ISSUE TITLE
Can't uncheck SSH server

### ISSUE THREAD

**Issue by weijiuqiao** [koreader/koreader#9812]
* KOReader version: 2022.10-122
* Device: kindle voyage

#### Issue
Cannot uncheck SSH server
#### Steps to reproduce
SSH server was checked, cannot uncheck to allow for login without password.
##### `crash.log` (if applicable)
>kill: can't kill pid xxx: No such process
>iptables: Bad rule (does a matching rule exist in that Chain?)

**Comment by hius07** [comment]
Cannot reproduce on my Voyage (firmware 5.12.4) with 2022.10-122.

**Comment by weijiuqiao** [comment]
Did a full restart of my kindle and it's good again. Don't know how I triggered it...

**Comment by NiLuJe** [comment]
Possibly it got OOM-killed?

**Comment by weijiuqiao** [comment]
Possibly. Restarting KOReader alone didn't help.

**Comment by NiLuJe** [comment]
Yeah, that sounds somewhat sensible ;).

**Comment by NiLuJe** [comment]
Short of rewriting the whole thing to either use start-stop-daemon and/or simply warning if the pid file is stale instead of using it as the single source of truth, sure, why not ;o).

**Comment by vrazo-2006** [comment]
Hi, I'm having this same issue but restarting isn't working. I also tried restarting koreader which also didn't work. I think I could still ssh into my kindle by adding a key to it but for some reason simple usbc connection doesn't allow me to transfer files and apps like OpenMTP won't recognize the kindle. Does any have any ides of how to get the ssh unticked or another way to get my key onto the kindle? Thanks for the help. Kindle paper white 11 2024. Looking closer I think the issue is when we check our pid in 
```lua
    local function isProcAlive(p)
        return p and util.pathExists("/proc/" .. p)
    end

``` 
we're not checking whether this pid actually belong to our dropbear ssh process. I suspect another process might have taken up that pid while the actual process HAS been terminated. Is there a way we can add a check to verify whether this process is our dropbear ssh? 

I was able to fix the issue by ticking the "Force close all connections on stop" then ticking off "SSH server"
